versioned_docs/version-1.0/cadence-migration-guide/usdc-migration.mdx

```diff
@@ -0,0 +1,336 @@
+It might also be useful to provide helpful error messages in functions and transactions
+that direct users to how they can swap their `FiatToken.Vault` for a `USDCFlow.Vault`.
+
+Example:
+```cadence
+// Old Code
+import FiatToken from 0xb19436aae4d94622
+
+pub fun sendFiatTokenToAddress(to: Address, vault: @FiatToken.Vault) {
+    // Get the recipient's public account object
+    let recipient = getAccount(to)
+
+    // Get a reference to the recipient's Receiver
+    let receiverRef = recipient.getCapability(FiatToken.VaultReceiverPubPath)
+        .borrow<&{FungibleToken.Receiver}>()
+        ?? panic("Could not borrow receiver reference to the recipient's Vault")
+
+    // Deposit the withdrawn tokens in the recipient's receiver
+    receiverRef.deposit(from: <-vault)
+}
+
+// New Code
+import USDCFlow from 0x{USDCFlowAddress}
+
+pub fun sendUSDCFlowToAddress(to: Address, vault: @USDCFlow.Vault) {
+    // Get the recipient's public account object
+    let recipient = getAccount(to)
+
+    // Get a reference to the recipient's Receiver
+    let receiverRef = recipient.capabilities.borrow<&{FungibleToken.Receiver}>(USDCFlow.ReceiverPublicPath)
+        ?? panic("Could not borrow receiver reference to the recipient's Vault")
+
+    // Deposit the withdrawn tokens in the recipient's receiver
+    receiverRef.deposit(from: <-vault)
+}
+```
+This does not apply to any contract with stored `FiatToken` values such as vaults though!
+If you store any `FiatToken` values, see the next section.
+
+## Stored `@FiatToken.Vault` Field
+
+If your contract stores ANY types from the `FiatToken` contract, 
+this means that you have state that will not work properly after the Crescendo upgrade.
+
+The only type from `FiatToken` that non-admins can store is `Vault`,
+so we will focus on that for now.
+
+### Contract-level Vault Field
+
+A contract-level field is a field that is stored in the contract state.
+For example, `FiatToken.totalSupply` is a contract field, but `FiatToken.Vault.balance`
+is not a contract field since it is defined on a resource that can live in other accounts.
+
+Here are some examples of `FiatToken.Vault` contract fields:
+```cadence
+pub contract VaultStorer {
+    // A Vault contract field
+    access(self) let vault: @FiatToken.Vault
+
+    // A Dictionary of Vaults contract field
+    access(self) let vaultDict: @{UInt64: FiatToken.Vault}
+
+    // An array of Vaults contract field
+    access(self) let vaults: @[FiatToken.Vault]
+}
+```
+
+After the Crescendo upgrade, you'll still be able to access the balance of these,
+but you won't be able to call any functions on them, including `deposit()` and `withdraw()`.
+This means that if your contract relies on any of that functionality,
+you'll need to do some extra work to migrate to the new token.
+
+The first thing to consider is if you are able to withdraw the USDC from these fields
+in any way. Ideally, you could withdraw the USDC from these `FiatToken` fields
```

Comment:
Lets change 'FiatToken' here to USDC because regular users won't understand FiatToken if they are coming just to this section to understand next steps.

Reply:
This is the developer section of the doc, so they should all understand what `FiatToken` is